runtime/doc/api-ui-events.txt

```diff
@@ -913,4 +932,29 @@ must handle.
 	prev_cmd
 	    True when sent with |g<| command, false with |:messages|.
 
+==============================================================================
+PROGRESS MESSAGE                                    *progress-message*
```

Comment:
This help file is for the UI protocol. Progress messages are not specific to the UI protocol (though they happen to emit UI events, just like any other regular message).

Add this as section "4." after this: https://github.com/neovim/neovim/blob/58060c2340a52377a0e1d2b782ce1deef13b2b9b/runtime/doc/message.txt#L754-L755